analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,357 @@
+  status_sym <- rlang::sym(status_variable)
+
+  status_plot <- ggplot2::ggplot(
+    status_count_df,
+    ggplot2::aes(
+      x = dominant_status,
+      y = !!status_sym
+    )
+  ) +
+    ggplot2::geom_point() +
+    ggplot2::ylab(status_variable)
+}
+```
+
+
+### Files and directories
+
+```{r}
+results_dir <- "results"
+
+# Define a logical object for running in CI
+running_in_ci <- params$is_ci
+```
+
+### Read in files
+
+```{r}
+# Read in the file with consensus CN status data and the UCSC cytoband data --
+# generated in `03-add-cytoband-status-consensus.Rmd`
+consensus_seg_cytoband_status_df <-
+  read_tsv(file.path("results", "consensus_seg_with_ucsc_cytoband_status.tsv.gz"))
+
+# Read in the annotated CN file (without the UCSC data)
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz"))
+```
+
+## Define most focal units
+
+### First, we want find entire chromosome arm losses/gains.
+
+```{r}
+consensus_seg_cytoband_status_df <-
+  consensus_seg_cytoband_status_df %>%
+  select(
+    chromosome_arm,
+    dominant_status,
+    cytoband,
+    Kids_First_Biospecimen_ID
+  )
+
+arm_status_count <- status_majority_caller(
+  consensus_seg_cytoband_status_df,
+  "chromosome_arm"
+)
+
+# These are the chromosome arms that have not been defined as gain or loss --
+# we want to define their cytoband/gene level status
+neutral_arm_status_df <- arm_status_count %>%
+  select(chromosome_arm, dominant_status) %>%
+  filter(dominant_status == "neutral")
+
+# Display table
+arm_status_count %>%
+  group_by(dominant_status)
+```
+
+### Next we want to use the annotated CN data to define the cytoband gains/losses 
+
+We want to include cytoband and gene level calls for chromosome arms that have not been defined as a gain or loss.
+
+```{r}
+# Filter the annotated CN data to include only these chromosome arms
+consensus_seg_autosomes_df_develop <- consensus_seg_autosomes_df %>%
+  select(biospecimen_id, gene_symbol, cytoband, status) %>%
+  mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+  filter(chromosome_arm %in% neutral_arm_status_df$chromosome_arm)
+
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 50% of the total counts are
+# for that particular status
+cytoband_status_count <- status_majority_caller(
+  consensus_seg_autosomes_df_develop,
+  "cytoband"
+)
+
+# These are the cytobands that have not been defined as gain or loss --
+# we want to define their gene level status
+neutral_cytoband_status_df <- cytoband_status_count %>%
+  select(cytoband, dominant_status) %>%
+  filter(dominant_status == "neutral")
+
+# Display table
+cytoband_status_count
+```
+
+### Now we want to use the annotated CN data to define the gene level gains/losses 
+
+```{r}
+# Filter the annotated CN data to include only these cytobands
+consensus_seg_autosomes_df_develop <- consensus_seg_autosomes_df %>%
```

Comment:
It looks like you are reusing the name `consensus_seg_autosomes_df_develop`a few times. This is a tad hard to follow. Can you make them different things? We should probably see if we can streamline this some, or add this formatting step to the main function.